### ISSUE TITLE
`terraform show` does not do a pull of remote state

### ISSUE THREAD

**Issue by rdgreen** [hashicorp/terraform#7050]
### Affected Resource(s)
- aws_ecs_cluster
### Expected Behavior

ws_ecs_cluster.default:
  id = arn:aws:ecs:eu-west-1:xxxxxxxxxxx:cluster/default
  name = default
### Actual Behaviour

No output
### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:
1. One one host, `terraform apply`
2. On another host, `terraform show`
### Important Factoids

tfstate is stored in S3.
### References

Possibly related to https://github.com/hashicorp/terraform/issues/1810

**Comment by jen20** [comment]
Hi @rdgreen! I think I agree that this is a bug, but need to chat with @phinze to confirm what the behaviour should be here. In the mean time you can do a `terraform pull` and then `terraform show` in order to get the latest state.

**Comment by phinze** [comment]
Hello! I agree that the behavior is surprising and I'm +1 for treating this like a bug, but this has been the behavior for quite some time now, so I'm going to remove the 0.7 tag here for now.

The `terraform remote pull; terraform show` combo is the proper workaround today.

The current behavior was set by past-me in the original implementation: https://github.com/hashicorp/terraform/pull/2371

Not a whole lot of extra context there - I think it was just an assumption that if you had remote state configured then it would also be pulled.

I think that fixing this is as `terraform show` do a `RefreshState()` call. I thought about adding some sort of `terraform show -pull` flag but I don't know if there's any reason to _not_ refresh state, and we can always add in that flag later for people to run `show` on just the remote cache if we get reports of actual use cases.

**Comment by apparentlymart** [comment]
Hi @rdgreen! Sorry for the problems here and sorry we let this languish for so long.

This issue did get fixed somewhere along the line, but I don't remember now exactly where. Most recently in version 0.9.0 we removed the concept of the local state cache altogether, so the remote state file is the only instance of the data for `terraform show` to pull from.

So while I can't be sure at what point this exact issue was fixed, I can say with some certainty that it will work as expected in 0.9 and above, and so I'm going to close this out. Thanks for raising this issue!
